src/lib/form/field.tsx

```diff
@@ -80,14 +84,67 @@ const StyledInput = styled.input<{
   padding-top: 14px;
   padding-bottom: 14px;
   padding-left: 16px;
-  padding-right: ${({ Icon, variant }) => {
-    if (Icon) return "56px";
-    if (variant) return "44px";
-    return "16px";
+  padding-right: ${({ Icon, variant, type }) => {
+    if (Icon) return type === "number" ? "64px" : "56px";
+    if (variant) return type === "number" ? "52px" : "44px";
+    return type === "number" ? "30px" : "16px";
   }};
+
+  /* Chrome, Safari, Edge, Opera */
+  ::-webkit-outer-spin-button,
+  ::-webkit-inner-spin-button {
+    -webkit-appearance: none;
+    margin: 0;
+  }
+
+  /* Firefox */
+  -moz-appearance: textfield;
+  appearance: textfield;
   ${baseInputStyle}
 `;
 
+const ArrowsContainer = styled.div<{
+  variant?: string;
+  Icon?: React.FC<React.SVGAttributes<SVGElement>>;
+}>`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  position: absolute;
+  height: 17px;
+  width: 14px;
+  top: 14px;
+  right: ${({ Icon, variant }) => {
+    if (Icon) return "48px";
+    if (variant) return "36px";
+    return "12px";
+  }};
+`;
+
+const ArrowButton = styled.button`
+  height: 10px;
+  width: 100%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  background: transparent;
+  border: none;
+  padding: 0;
```

Comment:
Here I would make the button square and add `border-radius` to go with the rest of the design
```suggestion
  height: 14px;
  width: 14px;
  display: flex;
  justify-content: center;
  align-items: center;
  background: transparent;
  border: none;
  border-radius: 3px;
  padding: 0;
```